guacamole-common/src/main/java/org/apache/guacamole/GuacamoleAuthenticationRejectedException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.apache.guacamole;
+
+/**
+ * An exception thrown when a successful authentication is rejected by a
+ * AuthenticationSuccessListener in an extension.
+ */
+public class GuacamoleAuthenticationRejectedException
```

Comment:
It was a judgement call. I subtyped existing exceptions to get status codes that should be appropriate for these cases, but was uncomfortable making assumptions about how the existing exception types might be interpreted in `catch` clauses. 

I'd argue that there is something fundamentally different about an API user providing an incorrect credential vs. an extension deciding that a successful authentication should be rejected. For that reason, I think a different exception type is appropriate. But I'm happy to change the code to reuse existing exceptions if that's preferred.